examples/huggingface/main.py

```diff
@@ -0,0 +1,320 @@
+#  [markdown]
+# # Fine-tune a pretrained model from Hugging Face
+#
+# source tutorial: https://huggingface.co/docs/transformers/training
+import logging
+import os
+
+os.environ["COMET_API_KEY"] = "comet_token"
+os.environ["COMET_WORKSPACE"] = "workspace"
+os.environ["COMET_PROJECT_NAME"] = "project"
+os.environ["COMET_MODE"] = "ONLINE"
+os.environ["COMET_LOG_ASSETS"] = "True"
+os.environ["COMET_AUTO_LOG_METRICS"] = "True"
+import argparse
+from copy import deepcopy
+
+import hydra
+import numpy as np
+import torch
+from datasets import load_dataset, load_metric
+from omegaconf import DictConfig
+from transformers import (
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer,
+    DataCollatorForSeq2Seq,
+    Seq2SeqTrainer,
+    Seq2SeqTrainingArguments,
+    TrainerCallback,
+)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-st",
+        "--size_train_dataset",
+        help="Number of samples to use from training data set. If not specified, use complete dataset",
+        type=int,
+        required=False,
+    )
+    parser.add_argument(
+        "-se",
+        "--size_eval_dataset",
+        help="Number of samples to use from evaluation data set. If not specified, use complete dataset",
+        type=int,
+        required=False,
+    )
+    parser.add_argument(
+        "-f",
+        "--freeze_base_model",
+        help="Freeze parameters of base model during training",
+        action="store_true",
+        required=False,
+    )
+    parser.add_argument(
+        "-lr", "--learning_rate", help="Learning rate", type=float, required=False
```

Comment:
Wait, does Hydra support arguments like `--learning-rate` or a we forced to use `_` instead? :worried:

Reply:
Actually, all of these arguments are sent to the `Seq2SeqTrainingArguments`. The problem since to be on huggingface's side. Looks like every argument uses `_` ... https://huggingface.co/docs/transformers/main_classes/trainer#transformers.Seq2SeqTrainingArguments

Reply:
It should be fine since `-` will be converted to `_`:
```python
>>> import argparse
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument('--some-arg')
>>> parser.parse_args(['--some-arg', 'hello'])
Namespace(some_arg='hello')
>>> vars(parser.parse_args(['--some-arg', 'hello']))
{'some_arg': 'hello'}

```